src/Kestrel.Transport.Sockets/Internal/SocketConnection.cs

```diff
@@ -263,8 +263,16 @@ private void Shutdown()
                     _aborted = true;
                     _trace.ConnectionWriteFin(ConnectionId);
 
-                    // Try to gracefully close the socket even for aborts to match libuv behavior.
-                    _socket.Shutdown(SocketShutdown.Both);
+                    try
+                    {
+                        // Try to gracefully close the socket even for aborts to match libuv behavior.
+                        _socket.Shutdown(SocketShutdown.Both);
+                    }
+                    catch
+                    {
+                        // Ignore any errors from Socket.Shutdown since we're tearing down the connection anyway.
+                    }
+
                     _socket.Dispose();
```

Comment:
@stephentoub @tmds I'm a little worried Socket.Dispose() could throw too. I don't wan't to put it in the same try block as Shutdown because we still want to call Dispose() when Shutdown throws. Should we add another try/catch or are we confident enough that Socket.Dispose() will never throw that we're willing to let the process crash if it does? Thanks.

Reply:
I've not seen exceptions emerge from Socket.Dispose, but that doesn't mean it couldn't happen.  @davidsh?

Reply:
Socket.Dispose() is never supposed to throw.  And if the Socket is already disposed, it should be a no-op.

Reply:
Thanks.

Reply:
Indeed. Socket.Dispose will never throw ODE. Only the first one to Dispose actually disposes and others noop: https://github.com/dotnet/corefx/blob/48ce3254e28f2dae2d9a5efe7dd42cfab119d287/src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs#L4456